src/ssl.c

```diff
@@ -166,6 +171,9 @@ WOLFSSL_CTX* wolfSSL_CTX_new(WOLFSSL_METHOD* method)
         if (ret != SSL_SUCCESS) {
             WOLFSSL_MSG("wolfSSL_Init failed");
             WOLFSSL_LEAVE("WOLFSSL_CTX_new", 0);
+            if (method != NULL) {
+                XFREE(method, NULL, DYNAMIC_TYPE_METHOD);
```

Comment:
Yes wolfSSL_CTX_free also frees the method. I'm not sure if that is an issue with the highlighted code here, but you are right that in the next XFREE around line 188 is not needed -- ctx will never be null in that case and wolfSSL_CTX_free will take care of freeing the method.